src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -148,10 +148,14 @@ public void visitToken(DetailAST ast) {
             leftCurly = slistToken;
         }
 
```

Comment:
@ps-sp , please do me a favor and move all that calculation of leftCurly to special method `final DetailAST leftCurly = findLeftCurly(ast)`
removing final is not allowed for `leftCurly` is only one left curly in code that we investigating and it can not change.